server/controllers/userController/register.ts

```diff
@@ -0,0 +1,71 @@
+import {Request, Response} from 'express';
+import {body} from 'express-validator';
+import httpStatus from 'http-status';
+import { MESSAGES } from 'consts';
+import { userService } from 'services';
+import { ArgumentValidationError, CustomError } from 'errors';
+import { errorHandlerWrapper } from 'utils/errorHandler.wrapper';
+import { Logger, encryptPassword } from 'utils';
+
+export const registerValidator = () => {
+    console.log("registered");
+    return[
+        body('email')
+            .notEmpty()
+            .withMessage({email: 'Email is required'})
+            .bail()
+            .isEmail()
+            .withMessage({email: 'Email is incorrect'}),
+        body('first_name')
+            .notEmpty()
+            .withMessage({first_name: 'First name is required'}),
+        body('last_name')
+            .notEmpty()
+            .withMessage({last_name: 'Last name is required'}),
+        body('password1')
+            .notEmpty()
+            .withMessage({password1: 'Password is required'}),
+        body('password2')
+            .notEmpty()
+            .withMessage({password2: 'Confirm password is required'}),
+    ];
+};
+
+type Params = unknown;
+type ResBody = unknown;
+type ReqBody = {
+    email?: string;
+    first_name?: string;
+    last_name?: string;
+    password1: string;
+    password2: string;
```

Comment:
what is the password1 and password2?

Reply:
password and confirm password.

Reply:
checking password and confirm password should be done in the frontend side.

Reply:
@TelmanWisley